flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/JoinTest.scala

```diff
@@ -0,0 +1,149 @@
+import org.apache.flink.table.utils.{StreamTableTestUtil, TableTestBase}
+import org.junit.Assert._
+import org.junit.Test
+
+class JoinTest extends TableTestBase {
+  private val streamUtil: StreamTableTestUtil = streamTestUtil()
+  streamUtil.addTable[(Int, String, Long)]("MyTable", 'a, 'b, 'c.rowtime, 'proctime.proctime)
+  streamUtil.addTable[(Int, String, Long)]("MyTable2", 'a, 'b, 'c.rowtime, 'proctime.proctime)
+
+  @Test
+  def testProcessingTimeInnerJoin() = {
+
+    val sqlQuery = "SELECT t1.a, t2.b " +
+      "FROM MyTable as t1 join MyTable2 as t2 on t1.a = t2.a and " +
+      "t1.proctime between t2.proctime - interval '1' hour and t2.proctime + interval '1' hour"
+    val expected =
+      unaryNode(
+        "DataStreamCalc",
+        binaryNode(
+          "DataStreamRowStreamJoin",
+          unaryNode(
+            "DataStreamCalc",
+            streamTableNode(0),
+            term("select", "a", "proctime")
+          ),
+          unaryNode(
+            "DataStreamCalc",
+            streamTableNode(1),
+            term("select", "a", "b", "proctime")
+          ),
+          term("condition",
+            "AND(=(a, a0), >=(TIME_MATERIALIZATION(proctime), " +
+              "-(TIME_MATERIALIZATION(proctime0), 3600000)), " +
+              "<=(TIME_MATERIALIZATION(proctime), " +
+              "DATETIME_PLUS(TIME_MATERIALIZATION(proctime0), 3600000)))"),
+          term("select", "a, proctime, a0, b, proctime0"),
+          term("joinType", "InnerJoin")
+        ),
+        term("select", "a", "b")
+      )
+
+    streamUtil.verifySql(sqlQuery, expected)
+  }
+
+
+  @Test
+  def testJoinTimeBoundary(): Unit = {
+    verifyTimeBoundary(
+      "t1.proctime between t2.proctime - interval '1' hour " +
+        "and t2.proctime + interval '1' hour",
+      3600000,
+      3600000,
+      "proctime")
+
+    verifyTimeBoundary(
+      "t1.proctime > t2.proctime - interval '1' second and " +
+        "t1.proctime < t2.proctime + interval '1' second",
+      999,
+      999,
+      "proctime")
+
+    verifyTimeBoundary(
+      "t1.c >= t2.c - interval '1' second and " +
+        "t1.c <= t2.c + interval '1' second",
+      1000,
+      1000,
+      "rowtime")
+
+    verifyTimeBoundary(
+      "t1.c >= t2.c and " +
+        "t1.c <= t2.c + interval '1' second",
+      0,
+      1000,
+      "rowtime")
+
+    verifyTimeBoundary(
+      "t1.c >= t2.c + interval '1' second and " +
+        "t1.c <= t2.c + interval '10' second",
+      0,
+      10000,
+      "rowtime")
+
+    verifyTimeBoundary(
+      "t2.c - interval '1' second <= t1.c and " +
+        "t2.c + interval '10' second >= t1.c",
+      1000,
+      10000,
+      "rowtime")
+
+    verifyTimeBoundary(
+      "t1.c - interval '2' second >= t2.c + interval '1' second -" +
+        "interval '10' second and " +
+        "t1.c <= t2.c + interval '10' second",
+      7000,
+      10000,
+      "rowtime")
+  }
+
+  def verifyTimeBoundary(
```

Comment:
Yes, it's better to check unsupported case here, but i think we still need to extend TableTestBase, cause we need to use the StreamTableTestUtil to do the sql parse.